src/Codebase/Infrastructure/SourceParser/SourceParser.php

```diff
@@ -63,6 +64,9 @@ public function parseSource(SourceParserRequestDto $codebaseRequestDto): Codebas
             }
             $codebaseSourceDto = $codebaseSourceDto->setType($type);
             foreach ($this->sourceParsers as $sourceParser) {
+                if ($type === SourceParserRequestDto::CORE_TYPE && $sourceParser->getExtension() === PhpParser::PARSER_EXTENSION) {
```

Comment:
In this case this condition can be avoided.

Reply:
We need it, the same reason that in previous comment